basex-core/src/main/java/org/basex/query/func/sql/SqlExecute.java

```diff
@@ -39,7 +38,7 @@ public Iter iter(final QueryContext qc) throws QueryException {
     if(!(obj instanceof Connection)) throw BXSQ_CONN_X.get(info, id);
     try {
       final Statement stmt = ((Connection) obj).createStatement();
-      return stmt.execute(query) ? iter(stmt, true) : Empty.ITER;
+      return stmt.execute(query) ? iter(stmt, true) : Int.get(stmt.getUpdateCount()).iter();
```

Comment:
Sounds completely reasonable! Thanks for trying 3 different RDBMS. I’ll additionally check SQLite, and I’ll merge your PR if everything works as expected.